relentless/potential/potential.py

```diff
@@ -391,14 +416,27 @@ def derivative(self, pair, param, r):
         r,deriv,scalar_r = self._zeros(r)
         if any(r < 0):
             raise ValueError('r cannot be negative')
+        if not isinstance(var, Variable):
+            raise TypeError('Parameter with respect to which to take the derivative must be a Variable.')
 
         # only evaluate at points inside [rmin,rmax], if specified
         flags = np.ones(r.shape[0], dtype=bool)
         if params['rmin'] is not False:
             flags[r < params['rmin']] = False
         if params['rmax'] is not False:
             flags[r > params['rmax']] = False
-        deriv[flags] = self._derivative(param, r[flags], **params)
+
+        # evaluate with respect to parameter
+        deriv[flags] = 0
+        d_params = [i for i in self.coeff.params if not(i=='rmin' or i=='rmax' or i=='shift')]
+        for p in d_params:
```

Comment:
We should use this PR to fix issues with the derivatives so that all parameters, including `rmin`, `rmax`, and `shift`, are correctly included (#8). It should be legal to take a derivative with respect to `rmin` and `rmax`, including chain rules, but we will need to use the `_force` method rather than the `_derivative` method. Also, `shift` is never going to be a variable (it is true/false), but we will need to include the effects of shifting too. I have some notes on this in a Mathematica notebook that we can finish working through.

For now, let's correct this loop to just run over `self.coeff.params`, and put in `if` statements that just `continue` the loop if `p` is 'rmin', 'rmax', or 'shift'.